### PR TITLE
Only one IV bag at a time

### PR DESCRIPTION
**Pull request by larentoun**
## What Does This PR Do

Now IV bags check if a target already have an IV bag, so only one can be inserted at a time.

Decided to use trait system instead of an element. Any suggestions welcomed.

Also not sure how good it looks it examine. I could've added a new signal for human examine, so it would've been displayed before HUD info.

## Why It's Good For The Game

Reduces powergame value of IV bags with huge amount of chems. Having 5 IVs from your backpack should be wrong.

## Images of changes

Screenshot_7

## Testing

* Applied IV bag
* Can't add IV bag
* Can remove IV bag
* Applied IV bag and deleted it
* Trait disappeared

## Changelog
:cl:
tweak: A human can only have a single IV bag injected into them at a time
/:cl:

### CODE REVIEW

File: code/modules/reagents/reagent_containers/iv_bag.dm [hunk 1]
```diff
@@ -41,11 +41,19 @@
 	..()
 	update_icon(UPDATE_OVERLAYS)
 
+/obj/item/reagent_containers/iv_bag/proc/on_examine(datum/source, mob/examiner, list/examine_list)
+	SIGNAL_HANDLER
```
Comment (DGamerL) [inline code/modules/reagents/reagent_containers/iv_bag.dm:45]:
```suggestion
	SIGNAL_HANDLER // COMSIG_PARENT_EXAMINE
```

**Review by DGamerL — COMMENTED** [review]
Add your trait in `traits.dm`

**Review by Sirryan2002 — APPROVED** [review]

**Comment by Spaghetti-bit** [comment]
A possibly unintended side effect is medical blood bags. Generally, doctors will insert multiple IVs of blood to speed up revival from fully drained bodies. I don't think doctors giving people two IVs of blood as power gaming. You have two arms. Possibly make it so you can have two IVs if you have one attached to an IV drip stand.

**Comment by Migratingcocofruit** [comment]
I think a better way to fix this would be to restrict how an IV is used rather than how many can be used. The main use of IVs is to fill up a drained crew member, and often this is done against the clock. This change hurts it a fair bit, and I can't really see much benefit.  

Is there any reason for targeting IV bags specifically? Because as far as I can see they don't offer too much of an advantage over pills, cigarettes/pipes, patches and so on.

**Review by lewcc — APPROVED** [review]

**Comment by larentoun** [comment]
> I think a better way to fix this would be to restrict how an IV is used rather than how many can be used. The main use of IVs is to fill up a drained crew member, and often this is done against the clock. This change hurts it a fair bit, and I can't really see much benefit.

Giving CPR extend the timer by up to 1 minute, so that's one way to fight the clock

**Pull request merged**